exercises/bracket-push/src/example.c

```diff
@@ -0,0 +1,50 @@
+#include "bracket_push.h"
+#include <stdio.h>
+
+static bool is_bbp(char c)
+{
+   return (c == '{') || (c == '}') ||
+       (c == '[') || (c == ']') || (c == '(') || (c == ')');
+}
+
+static bool matching(char a, char b)
+{
+   if (a == '}')
+      return b == '{';
+   if (a == ']')
+      return b == '[';
+   if (a == ')')
+      return b == '(';
+
+   return false;
+}
+
+bool bbp_okay(const char *input)
+{
+   if (input == NULL || *input == '\0') {
+      return true;
+   }
+
+   char stack[MAXBBP] = { 0 };
```

Comment:
I don't know that it's worth it, but you could use the C call stack as your stack and not need to allocate a fixed length array that might not be big enough. This would use a lot more memory, though :)

Reply:
I am afraid I can not follow - can you please explain a bit more? This sounds interesting

Reply:
Because I think this is interesting, I made a solution in Lua. If you don't know Lua but know JavaScript you will probably be able to understand the logic. You can view it here if you want to check it out:
http://exercism.io/submissions/b4bff072c1b84b7baba62e1e9c2de7ad

Reply:
The idea was indeed very interesting. Thanks.

I updated the example.c to use recursion. It is not mine solution (see link) but I like it better.